src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java

```diff
@@ -0,0 +1,112 @@
+import org.geoserver.ogcapi.APIDispatcher;
+import org.geoserver.ogcapi.APIService;
+import org.geoserver.ogcapi.ConformanceDocument;
+import org.geoserver.ogcapi.HTMLResponseBody;
+import org.geoserver.ogcapi.OpenAPIMessageConverter;
+import org.geoserver.opensearch.eo.OSEOInfo;
+import org.geoserver.opensearch.eo.OpenSearchAccessProvider;
+import org.geotools.util.logging.Logging;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/** Implementation of OGC Features API service */
+@APIService(
+    service = "STAC",
+    version = "1.0",
+    landingPage = "ogc/stac",
+    serviceClass = OSEOInfo.class
+)
+@RequestMapping(path = APIDispatcher.ROOT_PATH + "/stac")
+public class STACService {
+
+    public static final String STAC_VERSION = "1.0.0-beta.2";
+
+    public static final String FEATURE_CORE =
+            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core";
+    public static final String FEATURE_HTML =
+            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/html";
+    public static final String FEATURE_GEOJSON =
+            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson";
+    public static final String FEATURE_OAS30 =
+            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30";
+    public static final String FEATURE_CQL_TEXT =
+            "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/x-cql-text";
+    public static final String STAC_CORE = "https://api.stacspec.org/v1.0.0-beta.1/core";
+    public static final String STAC_SEARCH = "https://api.stacspec.org/v1.0.0-beta.1/item-search";
+    public static final String STAC_FEATURES =
+            "https://api.stacspec.org/spec/v1.0.0-beta.1/ogcapi-features";
+
+    private static final String DISPLAY_NAME = "SpatioTemporal Asset Catalog";
+
+    static final Logger LOGGER = Logging.getLogger(STACService.class);
+
+    private final GeoServer geoServer;
+    private final OpenSearchAccessProvider accessProvider;
+
+    public STACService(GeoServer geoServer, OpenSearchAccessProvider accessProvider) {
+        this.geoServer = geoServer;
+        this.accessProvider = accessProvider;
+    }
+
+    public OSEOInfo getService() {
+        return geoServer.getService(OSEOInfo.class);
+    }
+
+    private Catalog getCatalog() {
+        return geoServer.getCatalog();
+    }
+
+    @GetMapping(name = "getLandingPage")
+    @ResponseBody
+    @HTMLResponseBody(templateName = "landingPage.ftl", fileName = "landingPage.html")
+    public STACLandingPage getLandingPage() throws IOException {
+        return new STACLandingPage(getService(), "ogc/stac");
+    }
+
+    @GetMapping(path = "conformance", name = "getConformanceDeclaration")
+    @ResponseBody
+    @HTMLResponseBody(templateName = "conformance.ftl", fileName = "conformance.html")
```

Comment:
Note that this stuff has to get repeated in the landing page too. 

https://github.com/radiantearth/stac-api-spec/tree/master/core#stac-api---core-specification :
"Note the conformsTo JSON object follows exactly the structure of OGC API - Features declaration of conformance classes, except is available directly under the landing page. This is a slight break from how OGC API does things, as STAC feels it is important for clients to be able to understand conformance in a single request. To be conformant to OGC API's the /conformance endpoint must be implemented as well."

Reply:
Oddball... but in the spec indeed. Fixed.

Reply:
Oddball enough I thought I should quote both reference and text! If I hadn't seen it, might not have believed it...